fdbserver/TenantCache.actor.cpp

```diff
@@ -116,6 +118,39 @@ class TenantCacheImpl {
 			}
 		}
 	}
+
+	ACTOR static Future<Void> monitorStorageUsage(TenantCache* tenantCache) {
+		TraceEvent(SevInfo, "StartingTenantCacheStorageUsageMonitor", tenantCache->id()).log();
+
+		state int refreshInterval = SERVER_KNOBS->TENANT_CACHE_STORAGE_REFRESH_INTERVAL;
+		state double lastTenantListFetchTime = now();
+
+		loop {
+			if (now() - lastTenantListFetchTime > (2 * refreshInterval)) {
+				TraceEvent(SevWarn, "TenantCacheGetStorageUsageRefreshDelay", tenantCache->id()).log();
+			}
+
+			state std::vector<std::pair<KeyRef, TenantName>> tenantList = tenantCache->getTenantList();
+			state int i;
+			for (i = 0; i < tenantList.size(); i++) {
+				state ReadYourWritesTransaction tr(tenantCache->dbcx(), tenantList[i].second);
+				try {
+					state int64_t size = wait(tr.getEstimatedRangeSizeBytes(normalKeys));
+					tenantCache->updateStorageUsage(tenantList[i].first, size);
+				} catch (Error& e) {
+					if (e.code() != error_code_actor_cancelled) {
+						TraceEvent("TenantCacheGetStorageUsageError", tenantCache->id())
+						    .errorUnsuppressed(e)
+						    .suppressFor(1.0);
+					}
+					wait(tr.onError(e));
+				}
```

Comment:
This should be put inside a `loop`. Otherwise retryable errors will not be retried.